pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -511,14 +511,22 @@ func SeedSettingOwnerChecksEnabled(settings *gardencorev1beta1.SeedSettings) boo
 	return settings == nil || settings.OwnerChecks == nil || settings.OwnerChecks.Enabled
 }
 
-// SeedSettingDependencyWatchdogEndpointEnabled returns true if the depedency-watchdog-endpoint is enabled.
-func SeedSettingDependencyWatchdogEndpointEnabled(settings *gardencorev1beta1.SeedSettings) bool {
-	return settings == nil || settings.DependencyWatchdog == nil || settings.DependencyWatchdog.Endpoint == nil || settings.DependencyWatchdog.Endpoint.Enabled
-}
-
-// SeedSettingDependencyWatchdogProbeEnabled returns true if the depedency-watchdog-probe is enabled.
-func SeedSettingDependencyWatchdogProbeEnabled(settings *gardencorev1beta1.SeedSettings) bool {
-	return settings == nil || settings.DependencyWatchdog == nil || settings.DependencyWatchdog.Probe == nil || settings.DependencyWatchdog.Probe.Enabled
+// SeedSettingDependencyWatchdogWeederEnabled returns true if the dependency-watchdog-weeder is enabled. By default, this will be enabled unless explicitly set to false in `settings.DependencyWatchdog.Weeder.Enabled` or `settings.DependencyWatchdog.Endpoint.Enabled`(deprecated).
+// TODO(himanshu-kun): Rely just on `settings.DependencyWatchdog.Weeder` field after few releases.
```

Comment:
You can rather make `gardener-apiserver` always keep `weeder/endpoint` and `probe/prober` in sync. Then all clients can just make use of the new fields and don't need to care about the deprecated old fields anymore.

Reply:
sure I'll try that.